service/src/main/java/org/mskcc/smile/cpt_gateway/service/impl/MessageHandlingServiceImpl.java

```diff
@@ -101,8 +125,16 @@ public void initialize(Gateway gateway) throws Exception {
         if (!initialized) {
             messagingGateway = gateway;
             setupCMONewRequestSubscriber(messagingGateway, this);
-            setupIGORequestStatusSubscriber(messagingGateway, this);
             initializeNewRequestHandlers();
+            if (CPT_UPDATE_REQUEST_RECORD_URL.length() > 0) {
+                setupCMOUpdateRequestSubscriber(messagingGateway, this);
+                initializeUpdateRequestHandlers();
+            }
+            if (CPT_UPDATE_SAMPLE_RECORD_URL.length() > 0) {
+                setupCMOUpdateSampleSubscriber(messagingGateway, this);
+                initializeUpdateSampleHandlers();
+            }
```

Comment:
a little confused by this - would it not always be the case that the cpt gateway app would have these prop's set?

or are these checks more of a transitional thing?

Reply:
Yeah, I didn't like this.  I did this because I didn't want to keep us from deploying until Shalabh creates URLs for us to post updates.  Having said this, I removed these checks and put a check in the method that does the posting to look for empty URLs.